src/v/redpanda/admin/server.cc

```diff
@@ -1159,6 +1160,15 @@ ss::future<> admin_server::throw_on_error(
             throw ss::httpd::server_error_exception(
               fmt::format("Unexpected error: {}", ec.message()));
         }
+    } else if (ec.category() == wasm::error_category()) {
```

Comment:
Could probalby use a ducktape test for the plumbing and whatnot

Reply:
Good call. Done